llvm/lib/Target/RISCV/RISCVOptWInstrs.cpp

```diff
@@ -711,7 +774,12 @@ bool RISCVOptWInstrs::runOnMachineFunction(MachineFunction &MF) {
 
   bool MadeChange = false;
   MadeChange |= removeSExtWInstrs(MF, TII, ST, MRI);
-  MadeChange |= stripWSuffixes(MF, TII, ST, MRI);
+
+  if (!(DisableStripWSuffix || ST.preferWInst()))
+    MadeChange |= stripWSuffixes(MF, TII, ST, MRI);
+
+  if (EnableAppendWSuffix || ST.preferWInst())
```

Comment:
`EnableAppendWSuffix` should not override the target preference. The command line option should only be for debug if the pass happens to introduce a bug. You can use -mattr=+prefer-w-inst for testing.